### PR TITLE
core: arm: add more cp15 macros

### PR DESCRIPTION
**Pull request by MrVan**
Add more cp15 macros.
These are useful when do suspend/resume operations in OP-TEE.

### CODE REVIEW

File: core/arch/arm/include/arm32_macros.S [hunk 1]
```diff
@@ -123,10 +188,30 @@
...
+	.macro write_tpidruro reg
+	mcr	p15, 0, \reg, c13, c0, 3
+	.endm
+
+	.macro read_tpidruro reg
+	mrc	p15, 0, \reg, c13, c0, 3
+	.endm
+
+	.macro write_diag reg
+	mcr  p15, 0, \reg, c15, c0, 1
```
Comment (jenswi-linaro) [inline core/arch/arm/include/arm32_macros.S]: All the `c15` registers are implementation defined, that is, defined based on CPU type.
Perhaps we can keep them in a `arm32_macros_cortex_a??.S` etc as needed.

Comment (MrVan) [inline core/arch/arm/include/arm32_macros.S]: @jenswi-linaro Agree. I'll fix it in V2.

**Comment by jenswi-linaro** [comment]
Sorry, didn't notice the update before.

`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by MrVan** [comment]
fix the build error and tags applied.

**Comment by jforissier** [comment]
Merged, thanks @MrVan!

**Pull request merged**